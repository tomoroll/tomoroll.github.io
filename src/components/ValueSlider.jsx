import React from 'react';
import Slider from 'rc-slider/lib/Slider';
import PropTypes from 'prop-types';
import 'rc-slider/assets/index.css';


function ValueSlider({
  value, updateValue, step, min, max, addonText, toFixedDigits,
}) {
  let addon = null;
  if (addonText !== null) {
    addon = (
      <div className="input-group-append">
        <span className="input-group-text text-monospace">{addonText}</span>
      </div>
    );
  }
  const formattedValue = toFixedDigits === null ? value : value.toFixed(toFixedDigits);
  return (
    <div className="row">
      <div className="input-group col-sm-3 col-lg-2">
        <input
          type="number"
          className="form-control"
          onChange={e => updateValue(Number(e.target.value))}
          value={formattedValue}
        />
        {addon}
      </div>
      <div className="col">
        <Slider onChange={updateValue} value={value} step={step} min={min} max={max} />
      </div>
    </div>
  );
}
ValueSlider.propTypes = {
  value: PropTypes.number.isRequired,
  updateValue: PropTypes.func.isRequired,
  step: PropTypes.number,
  min: PropTypes.number,
  max: PropTypes.number,
  addonText: PropTypes.string,
  toFixedDigits: PropTypes.number,
};
ValueSlider.defaultProps = {
  step: 1,
  min: 0,
  max: 99,
  addonText: null,
  toFixedDigits: null,
};

export default ValueSlider;
