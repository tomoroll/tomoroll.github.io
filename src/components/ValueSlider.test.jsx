import React from 'react';
import renderer from 'react-test-renderer';
import ValueSlider from './ValueSlider';

it('renders correctly', () => {
  const tree = renderer
    .create(<ValueSlider value={10} updateValue={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
