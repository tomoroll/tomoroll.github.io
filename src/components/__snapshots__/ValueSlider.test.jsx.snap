// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="input-group col-sm-3 col-lg-2"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="number"
      value={10}
    />
  </div>
  <div
    className="col"
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "width": "10%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={10}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "10%",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;
